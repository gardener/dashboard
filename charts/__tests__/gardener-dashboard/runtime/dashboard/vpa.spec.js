//
// SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and Gardener contributors
//
// SPDX-License-Identifier: Apache-2.0
//

'use strict'

const { helm } = fixtures

const renderTemplates = helm.renderDashboardRuntimeTemplates

describe('gardener-dashboard', function () {
  describe('vpa', function () {
    let templates

    beforeEach(() => {
      templates = [
        'vpa'
      ]
    })

    it('should render the template with default values', async function () {
      const values = {
        global: {
          dashboard: {
            vpa: {
              dummy: false
            }
          }
        }
      }
      const documents = await renderTemplates(templates, values)
      expect(documents).toHaveLength(1)
      const [vpa] = documents
      expect(vpa).toMatchSnapshot()
    })

    it('should not render the template', async function () {
      const values = {}
      const documents = await renderTemplates(templates, values)
      expect(documents).toHaveLength(1)
      const [vpa] = documents
      expect(vpa).toBe(null)
    })

    it('should render the template with overwritten values', async function () {
      const values = {
        global: {
          dashboard: {
            vpa: {
              updateMode: 'Auto',
              controlledValues: 'RequestsAndLimits',
              minAllowedCpu: '30m',
              minAllowedMemory: '60Mi',
              maxAllowedCpu: '1000m',
              maxAllowedMemory: '1200Mi'
            }
          }
        }
      }
      const documents = await renderTemplates(templates, values)
      expect(documents).toHaveLength(1)
      const [vpa] = documents
      expect(vpa).toMatchSnapshot()
    })
  })
})
