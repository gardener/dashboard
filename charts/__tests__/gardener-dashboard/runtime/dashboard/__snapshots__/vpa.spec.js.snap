// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gardener-dashboard vpa should render the template with default values 1`] = `
{
  "apiVersion": "autoscaling.k8s.io/v1",
  "kind": "VerticalPodAutoscaler",
  "metadata": {
    "labels": {
      "app.kubernetes.io/component": "dashboard",
      "app.kubernetes.io/instance": "gardener-dashboard",
      "app.kubernetes.io/managed-by": "Helm",
      "app.kubernetes.io/name": "gardener-dashboard",
      "helm.sh/chart": "gardener-dashboard-runtime-0.1.0",
    },
    "name": "gardener-dashboard-vpa",
    "namespace": "garden",
  },
  "spec": {
    "resourcePolicy": {
      "containerPolicies": [
        {
          "containerName": "*",
          "controlledResources": [
            "cpu",
            "memory",
          ],
          "controlledValues": "RequestsOnly",
          "maxAllowed": {
            "cpu": "500m",
            "memory": "500Mi",
          },
          "minAllowed": {
            "cpu": "50m",
            "memory": "100Mi",
          },
        },
      ],
    },
    "targetRef": {
      "apiVersion": "apps/v1",
      "kind": "Deployment",
      "name": "gardener-dashboard",
    },
    "updatePolicy": {
      "updateMode": "Off",
    },
  },
}
`;

exports[`gardener-dashboard vpa should render the template with overwritten values 1`] = `
{
  "apiVersion": "autoscaling.k8s.io/v1",
  "kind": "VerticalPodAutoscaler",
  "metadata": {
    "labels": {
      "app.kubernetes.io/component": "dashboard",
      "app.kubernetes.io/instance": "gardener-dashboard",
      "app.kubernetes.io/managed-by": "Helm",
      "app.kubernetes.io/name": "gardener-dashboard",
      "helm.sh/chart": "gardener-dashboard-runtime-0.1.0",
    },
    "name": "gardener-dashboard-vpa",
    "namespace": "garden",
  },
  "spec": {
    "resourcePolicy": {
      "containerPolicies": [
        {
          "containerName": "*",
          "controlledResources": [
            "cpu",
            "memory",
          ],
          "controlledValues": "RequestsAndLimits",
          "maxAllowed": {
            "cpu": "1000m",
            "memory": "1200Mi",
          },
          "minAllowed": {
            "cpu": "30m",
            "memory": "60Mi",
          },
        },
      ],
    },
    "targetRef": {
      "apiVersion": "apps/v1",
      "kind": "Deployment",
      "name": "gardener-dashboard",
    },
    "updatePolicy": {
      "updateMode": "Auto",
    },
  },
}
`;
